daemon/src/taker.rs

```diff
@@ -292,3 +242,93 @@ async fn main() -> Result<()> {
 
     Ok(())
 }
+
+pub struct ActorSystem<O, M> {
+    cfd_actor_addr: Address<taker_cfd::Actor<O, M>>,
+    cfd_feed_receiver: watch::Receiver<Vec<Cfd>>,
+    order_feed_receiver: watch::Receiver<Option<Order>>,
+    update_cfd_feed_receiver: watch::Receiver<UpdateCfdProposals>,
+}
+
+impl<O, M> ActorSystem<O, M>
+where
+    O: xtra::Handler<oracle::MonitorAttestation>
+        + xtra::Handler<oracle::GetAnnouncement>
+        + xtra::Handler<oracle::FetchAnnouncement>
+        + xtra::Handler<oracle::Sync>,
+    M: xtra::Handler<monitor::StartMonitoring>
+        + xtra::Handler<monitor::Sync>
+        + xtra::Handler<monitor::CollaborativeSettlement>,
+{
+    pub async fn new<F>(
+        db: SqlitePool,
+        wallet: Wallet,
+        oracle_pk: schnorrsig::PublicKey,
+        send_to_maker: Box<dyn MessageChannel<wire::TakerToMaker>>,
+        read_from_maker: Box<dyn Stream<Item = taker_cfd::MakerStreamMessage> + Unpin + Send>,
+        oracle_constructor: impl Fn(Vec<Cfd>, Box<dyn StrongMessageChannel<Attestation>>) -> O,
+        monitor_constructor: impl Fn(Box<dyn StrongMessageChannel<monitor::Event>>, Vec<Cfd>) -> F,
```

Comment:
For the tests, we might need control over the oracle and monitor actors so you will likely need to return their addresses (`O` and `F`) from this function!

Reply:
> Found a bug and noted some potential improvements :)
> 
> Thanks!

This ActorSystem closely matches the one on the maker side - perhaps it would be best for me to create a follow-up that rolls out the suggestions that apply to both sides? (I'll address the ones that solely matter for the taker here)